v2/lib/gsmSecret.libsonnet

```diff
@@ -0,0 +1,41 @@
+local v = import '../internal/validation.libsonnet';
+{
+  withExternalSecret(name, secrets=[], allKeysFrom=[])::
```

Comment:
I'm unsure about the naming here. How can a user know the difference between `withSecret` and `withExternalSecret`? One is app and objects, the other is not.